src/core/native/browser_adapter.ts

```diff
@@ -121,7 +122,10 @@ export class BrowserAdapter implements Adapter {
 
   reload(reason: ReloadReason) {
     dispatch("turbo:reload", { detail: reason })
-    window.location.reload()
+
+    if (!this.location) return
+
+    window.location.href = this.location.toString()
```

Comment:
Is there any potential that this causes issues with or prevents reloads for same-page anchors? I'm not clear if that situation is possible here.

Reply:
I can't imagine a case where a same-page anchor causes a full page reload